part4/006.js

```diff
@@ -3,25 +3,26 @@
  */
 
 function solution(N, stages) {
-  const failureRate = {};
+  const failureRates = [];
 
-  let total = stages.length;
-  for (let i = 1; i <= N; i++) {
-    let count = 0;
-    for (let j = 0; j < stages.length; j++) {
-      if (i === stages[j]) count++;
-    }
-    failureRate[i] = count / total;
-    total = total - count;
-    count = 0;
-  }
+  const counts = Array.from({ length: N }).fill(0);
+  stages.forEach((currentStage) => {
+    if (currentStage - 1 >= N) return;
+    counts[currentStage - 1]++;
+  });
 
-  const sorted = Object.entries(failureRate).sort((a, b) => {
-    if (a[1] === b[1]) return a[0] - b[0];
-    return b[1] - a[1];
+  let total = stages.length;
+  counts.forEach((count, index) => {
+    failureRates.push([index + 1, count / total]);
+    total -= count;
   });
 
-  return sorted.map((item) => Number(item[0]));
+  return failureRates
+    .sort((a, b) => {
+      if (a[1] === b[1]) return a[0] - b[0];
+      return b[1] - a[1];
```

Comment:
위 조건을 좀 더 간단하게 정리할 수 있음

```suggestion
 .sort((a, b) => b[1] - a[1] ||  a[0] - b[0] )
```
> 실패율의 내림차순이거나 실패율이 같다면(b[1] - a[1] 가 false인 경우) stage의 오름차순으로 정렬